python/ct/crypto/in_memory_merkle_tree_test.py

```diff
@@ -0,0 +1,193 @@
+    ))]
+
+PRECOMPUTED_PROOF_TEST_VECTORS = [
+    ConsistencyTestVector(1, 1, []),
+    ConsistencyTestVector(1, 8, decode_hex_strings_list(
+            ["96a296d224f285c67bee93c30f8a309157f0daa35dc5b87e410b78630a09cfc7",
+             "5f083f0a1a33ca076a95279832580db3e0ef4584bdff1f54c8a360f50de3031e",
+             "6b47aaf29ee3c2af9af889bc1fb9254dabd31177f16232dd6aab035ca39bf6e4"]
+    )),
+    ConsistencyTestVector(6, 8, decode_hex_strings_list(
+            ["0ebc5d3437fbe2db158b9f126a1d118e308181031d0a949f8dededebc558ef6a",
+             "ca854ea128ed050b41b35ffc1b87b8eb2bde461e9e3b5596ece6b9d5975a0ae0",
+             "d37ee418976dd95753c1c73862b9398fa2a2cf9b4ff0fdfe8b30cd95209614b7"]
+    )),
+    ConsistencyTestVector(2, 5, decode_hex_strings_list(
+            ["5f083f0a1a33ca076a95279832580db3e0ef4584bdff1f54c8a360f50de3031e",
+             "bc1a0643b12e4d2d7c77918f44e0f4f79a838b6cf9ec5b5c283e1f4d88599e6b"]
+    )),
+    ]
+
+class InMemoryMerkleTreeTest(unittest.TestCase):
+    """Tests for InMemoryMerkleTree."""
+
+    def test_tree_incremental_root_hash(self):
+        """Test root hash calculated correctly when leaves are added
+        incrementally."""
+        tree = in_memory_merkle_tree.InMemoryMerkleTree([])
+        hasher = merkle.TreeHasher()
+        for i in range(len(TEST_VECTOR_DATA)):
+            tree.add_leaf(TEST_VECTOR_DATA[i])
+            self.assertEqual(
+                    tree.get_root_hash(),
+                    hasher.hash_full_tree(TEST_VECTOR_DATA[0:i+1]))
+
+    def test_tree_snapshot_root_hash(self):
+        """Test root hash calculated correctly when all leaves are added at
+        once."""
+        tree = in_memory_merkle_tree.InMemoryMerkleTree(TEST_VECTOR_DATA)
+        hasher = merkle.TreeHasher()
+        for i in range(len(TEST_VECTOR_DATA)):
+            self.assertEqual(
+                    tree.get_root_hash(i),
+                    hasher.hash_full_tree(TEST_VECTOR_DATA[0:i]))
+
+    def test_tree_inclusion_proof_precomputed(self):
+        """Inclusion proof generation correctness test for known-good
+        proofs."""
+        tree = in_memory_merkle_tree.InMemoryMerkleTree(TEST_VECTOR_DATA)
+        verifier = merkle.MerkleVerifier()
+        for v in PRECOMPUTED_PATH_TEST_VECTORS:
+            audit_path = tree.get_inclusion_proof(v.leaf, v.tree_size_snapshot)
+            self.assertEqual(len(audit_path), v.path_length)
+            self.assertEqual(audit_path, v.path)
+
+            leaf_data = TEST_VECTOR_DATA[v.leaf]
+            leaf_hash = merkle.TreeHasher().hash_leaf(leaf_data)
+            dummy_sth = DummySTH(v.tree_size_snapshot,
+                tree.get_root_hash(v.tree_size_snapshot))
+
+            if v.tree_size_snapshot > 0:
+                verifier.verify_leaf_hash_inclusion(
+                        leaf_hash, v.leaf, audit_path, dummy_sth)
+
+    def test_tree_inclusion_proof_generated(self):
```

Comment:
These "generated" tests add significantly to test duration (30-60 seconds). Do you think there's an argument for moving them out to a separate test file, so they can be left out of some test runs?